core/src/test/java/org/infinispan/notifications/cachelistener/CacheNotifierImplWithConverterTest.java

```diff
@@ -1,322 +1,41 @@
-      assert ((TransactionCompletedEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
-      assert cl.getEvents().get(1).getCache() == mockCache;
-      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_COMPLETED;
-      assert !((TransactionCompletedEvent) cl.getEvents().get(1)).isTransactionSuccessful();
-      assert ((TransactionCompletedEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
-   }
-
-   public void testNotifyTransactionRegistered() {
-      GlobalTransaction tx = mock(GlobalTransaction.class);
-      n.notifyTransactionRegistered(tx, false);
-      n.notifyTransactionRegistered(tx, false);
-
-      assert cl.getInvocationCount() == 2;
-      assert cl.getEvents().get(0).getCache() == mockCache;
-      assert cl.getEvents().get(0).getType() == Event.Type.TRANSACTION_REGISTERED;
-      assert ((TransactionRegisteredEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
-      assert cl.getEvents().get(1).getCache() == mockCache;
-      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_REGISTERED;
-      assert ((TransactionRegisteredEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
-   }
-   
-   @NamedFactory(name = "to-string-converter-factory")
-   public class ToStringEventConverterFactory implements CacheEventConverterFactory {
-
-   	@Override
-   	public CacheEventConverter<String, String, String> getConverter(Object[] params) {
-   		// TODO Auto-generated method stub
-   		return new ToStringConverter();
-   	}
-   	class ToStringConverter implements CacheEventConverter<String, String, String>  {
-   		      public String convert(String key, String oldValue, Metadata oldMetadata, 
-   		               String newValue, Metadata newMetadata, EventType eventType) {
-   		         return "custom event for test ("+(eventType==null ? "null" : eventType.getType().name())+","+ key+","+newValue+")";
-   		      }
-   		}
-
-   }
+public class CacheNotifierImplWithConverterTest extends CacheNotifierImplTest {
+
+	@Override
+	protected void addListener() {
+		// Installing a listener with converter
+		HashSet<Class> filterAnnotations = new HashSet<>(Arrays.asList(CacheEntryCreated.class,
+				CacheEntryModified.class, CacheEntryRemoved.class, CacheEntryExpired.class));
+		n.addFilteredListener(cl, null, new ToStringConverter(), filterAnnotations);
+	}
+
+	@Override
+	protected boolean isReturnedEventValueCorrect(CacheEntryEvent ev, Object key, Object val, Event.Type t) {
+		// ToStringConverter (below) returns a string with format:
+		// "custom event for test (CACHE_ENTRY_<event_type>,<key>,<value>)"
+		return ev.getValue().equals("custom event for test (" + t.name() + "," + key + "," + val + ")");
+	}
+
+	class ToStringConverter implements CacheEventConverter<String, String, String> {
+		public String convert(String key, String oldValue, Metadata oldMetadata, String newValue, Metadata newMetadata,
```

Comment:
Cool thing is you can use this as a Functional interface and provide a lambda to line 25 above.

`n.addFilteredListener(cl, null, (key, oldValue, oldMetadata, newValue, newMetadata, eventType) -> "custom event for test (" + eventType.getType().name() + "," + key + "," + newValue + ")", filterAnnotations);